theano/gpuarray/tests/test_ctc.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import (division, absolute_import, print_function)
+
+import unittest
+import numpy as np
+
+import theano
+import theano.tensor as T
+from theano.tests import unittest_tools as utt
+import theano.gpuarray
+from theano.gpuarray.ctc import (ctc_enabled, ctc)
+
+
+class TestCTC(unittest.TestCase):
+    def setUp(self):
+        if not ctc_enabled:
+            self.skipTest('Optional library warp-ctc not available')
+
+    def run_ctc(self, activations, labels, input_length, expected_costs, expected_grads):
+        # Create symbolic variables
+        t_activations = theano.shared(activations, name="activations")
+        t_activation_times = theano.shared(input_length, name="activation_times")
+        t_labels = theano.shared(labels, name="labels")
+
+        t_cost = ctc(t_activations, t_labels, t_activation_times)
+        # Symbolic gradient of CTC cost
+        t_grad = T.grad(T.mean(t_cost), t_activations)
+        # Compile symbolic functions
+        train = theano.function([], [t_cost, t_grad])
+
+        cost, grad = train()
+
+        cpu_cost = np.empty(shape=cost.shape, dtype=np.float32)
+        # Transfer costs from GPU memory to host
+        cost.read(cpu_cost)
+        cost.sync()
```

Comment:
You can just do `cpu_cost = np.asarray(cost)` here, without the sequence `np.empty, read, sync`. It should work.

Same comment for `gpu_grad` below.